### ISSUE TITLE
[gradle] 'outputDirectory' specifies directory which doesn't exist.

### ISSUE THREAD

**Issue by marceloverdijk** [jreleaser/jreleaser#1292]
Upgrading form JReleaser 1.3.1 to 1.5.1 using the Grade plugin I get:

```
* What went wrong:
A problem was found with the configuration of task ':jreleaserFullRelease' (type 'JReleaserFullReleaseTask').
  - In plugin 'org.jreleaser' type 'org.jreleaser.gradle.plugin.tasks.JReleaserFullReleaseTask' property 'outputDirectory' specifies directory '/home/runner/work/onlyf1-db/onlyf1-db/build/jreleaser' which doesn't exist.
    
    Reason: An input file was expected to be present but it doesn't exist.
    
    Possible solutions:
      1. Make sure the directory exists before the task is called.
      2. Make sure that the task which produces the directory is declared as an input.
    
    Please refer to https://docs.gradle.org/7.6/userguide/validation_problems.html#input_file_does_not_exist for more details about this problem.

```

Note this is a different error that I got before:
[Gradle 8: property 'outputDirectory' has @Input annotation used on property of type 'DirectoryProperty'](https://github.com/jreleaser/jreleaser/issues/1212) which was resolved and I wanted to test with 1.5.1.

I also cannot explain why it is looking for directory '/home/runner/work/onlyf1-db/onlyf1-db/build/jreleaser'; the 'jreleaser' subfolder I have not set anywhere myself.

Not sure if this is misconfig from my side...

Ahh, while typing this I remember that the previous issue (#1212) I have mentioned above was related to combination of Gradle 8 and JReleaser 1.4.x. But what I actually now did is use 1.5.1 with Gradle 7. So not completely comparable by still the output directory issue is not what I would expect.

_Originally posted by @marceloverdijk in https://github.com/jreleaser/jreleaser/discussions/1283_

**Comment by aalmiray** [comment]
I'm unable to reproduce the error running both Gradle `7.6` and `8.0.1`

```
$ ls -l build
drwxr-xr-x - aalmiray 24 Mar 12:00 -I distributions

$ gradle jreleaserFullRelease --no-build-cache
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources UP-TO-DATE
> Task :buildSrc:generateJsonSchema2Pojo UP-TO-DATE
> Task :buildSrc:kaptGenerateStubsKotlin UP-TO-DATE
> Task :buildSrc:kaptKotlin UP-TO-DATE
> Task :buildSrc:compileKotlin UP-TO-DATE
> Task :buildSrc:compileJava UP-TO-DATE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE

> Configure project :
JReleaser Gradle Plugin 1.5.1. Consider becoming a sponsor at https://opencollective.com/jreleaser

> Task :distTar SKIPPED
> Task :distZip NO-SOURCE
> Task :assembleDist UP-TO-DATE

> Task :jreleaserFullRelease
[INFO]  JReleaser 1.5.1
[INFO]    - basedir set to /private/tmp/onlyf1-db
[INFO]    - outputdir set to /private/tmp/onlyf1-db/build/jreleaser
[INFO]  git-root-search set to false
[INFO]  Loading variables from /Users/aalmiray/.jreleaser/config.toml
[INFO]  Validating configuration
[INFO]  Strict mode set to false
[INFO]  Project version set to 2023.2.0
[INFO]  Release is not snapshot
[INFO]  Timestamp is 2023-03-24T12:15:42.037736+01:00
[INFO]  HEAD is at 74dad93
[INFO]  Platform is osx-x86_64
[INFO]  dry-run set to true
[INFO]  Generating changelog
[INFO]  Storing changelog: build/jreleaser/release/CHANGELOG.md
[INFO]  Cataloging artifacts
[INFO]    [sbom] Cataloging is not enabled. Skipping
[INFO]  Calculating checksums for distributions and files
[INFO]    [checksum] build/distributions/onlyf1-db-json-single-2023.2.0.zip.sha256
[INFO]    [checksum] build/distributions/onlyf1-db-json-splitted-2023.2.0.zip.sha256
[INFO]    [checksum] build/distributions/onlyf1-db-smile-single-2023.2.0.zip.sha256
[INFO]    [checksum] build/distributions/onlyf1-db-smile-splitted-2023.2.0.zip.sha256
[INFO]  Signing distributions and files
[INFO]    [sign] Signing is not enabled. Skipping
[INFO]  Deploying Maven artifacts
[INFO]    [maven] Deploying is not enabled. Skipping
[INFO]  Uploading distributions and files
[INFO]    [upload] Uploading is not enabled. Skipping
[INFO]  Releasing to https://github.com/onlyf1com/onlyf1-db
[INFO]   - uploading checksums_sha256.txt
[INFO]   - uploading onlyf1-db-json-single-2023.2.0.zip
[INFO]   - uploading onlyf1-db-json-splitted-2023.2.0.zip
[INFO]   - uploading onlyf1-db-smile-single-2023.2.0.zip
[INFO]   - uploading onlyf1-db-smile-splitted-2023.2.0.zip
[INFO]  Announcing release
[INFO]    [announce] Announcing is not enabled. Skipping
[INFO]  Writing output properties to build/jreleaser/output.properties
[INFO]  JReleaser succeeded after 0.795 s

BUILD SUCCESSFUL in 7s
10 actionable tasks: 1 executed, 9 up-to-date
```

**Comment by marceloverdijk** [comment]
Hmmm,

I tried a dry-run locally and that gave no problem either:

```
❯ ./gradlew clean dist assemble jreleaserFullRelease --no-daemon --dry-run
To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.6/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build
:clean SKIPPED
:dist SKIPPED
:csvDistTar SKIPPED
:csvDistZip SKIPPED
:distTar SKIPPED
:distZip SKIPPED
:json-singleDistTar SKIPPED
:json-singleDistZip SKIPPED
:json-splittedDistTar SKIPPED
:json-splittedDistZip SKIPPED
:smile-singleDistTar SKIPPED
:smile-singleDistZip SKIPPED
:smile-splittedDistTar SKIPPED
:smile-splittedDistZip SKIPPED
:sqliteDistTar SKIPPED
:sqliteDistZip SKIPPED
:assemble SKIPPED
:assembleDist SKIPPED
:jreleaserFullRelease SKIPPED
```

There error I posted is from GitHub Action, could that make a difference?

**Comment by aalmiray** [comment]
I **can** replicate the error with the following command as seen in the GH action

```
./gradlew clean dist assemble jreleaserFullRelease --no-daemon
```

I believe the problem is the use of `clean` as that will delete the `build/jreleaser` directory but the logger remains active as a build service. solution: do not invoke `clean`, specially on a freshly checked out project.

**Comment by aalmiray** [comment]
More details. Logger is created as a build service 

https://github.com/jreleaser/jreleaser/blob/10742852a58603b852c6debafb9d50034d956b58/plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/internal/JReleaserProjectConfigurer.groovy#L73-L78

Windows crashes when logger is still open, thus it must be closed during clean

https://github.com/jreleaser/jreleaser/blob/10742852a58603b852c6debafb9d50034d956b58/plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/internal/JReleaserProjectConfigurer.groovy#L80-L90

Because the logger is set as a build service it's treated as a singleton and initialized only once. `clean` will delete the `build/jreleaser` directory but the logger has no clue about that.

**Comment by haydenbaker** [comment]
FWIW, I run into this issue when running any JReleaser tasks (JReleaser-1.10.0 and Gradle-8.2.1). The only workaround is to create the dir before any tasks run:
```
tasks.assembleDist.doFirst {
    mkdir "$buildDir/jreleaser"
}
```

**Comment by aalmiray** [comment]
Is the `clean` task invoked in the same execution session as a jreleaser task?

**Comment by Mahoney** [comment]
I also see this when running only `./gradlew -Pversion=1.0.0 jreleaserAssemble` (no `clean`).

Gradle 8.12.1, JReleaser 1.16.0 .

```
$ ./gradlew -Pversion=1.0.0 jreleaserAssemble 
Type-safe project accessors is an incubating feature.
> Task :jreleaserAssemble FAILED

[Incubating] Problems report is available at: file:///Users/robert/dev/wiremock/wiremock-cli/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':jreleaserAssemble' (type 'JReleaserAssembleTask').
  - In plugin 'org.jreleaser' type 'org.jreleaser.gradle.plugin.tasks.JReleaserAssembleTask' property 'outputDirectory' specifies directory '/Users/robert/dev/wiremock/wiremock-cli/build/jreleaser' which doesn't exist.
    
    Reason: An input file was expected to be present but it doesn't exist.
    
    Possible solutions:
      1. Make sure the directory exists before the task is called.
      2. Make sure that the task which produces the directory is declared as an input.
    
    For more information, please refer to https://docs.gradle.org/8.12.1/userguide/validation_problems.html#input_file_does_not_exist in the Gradle documentation.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 744ms
10 actionable tasks: 1 executed, 9 up-to-date

```

**Comment by aalmiray** [comment]
@Mahoney please open a new issue referencing this. There may be some additional work to be done to fix it.
